### PR TITLE
NO-JIRA: add check to ensure clusterprofile matches for featuregates

### PR DESCRIPTION
**Pull request by deads2k**
This should be a safe first step towards adding clusterprofiles to openshfit/api's payload manifest.

/assign @JoelSpeed

### CODE REVIEW

**Comment by deads2k** [comment]
testing with clusterbot `launch 4.16,openshift/api#1788,openshift/cluster-config-operator#411 aws`

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -214,10 +215,27 @@ func (o *OperatorOptions) RunOperator(ctx context.Context, controllerContext *co
 	return nil
 }
 
-func (o *OperatorOptions) getFeatureGateMappingFromDisk() (map[configv1.FeatureSet]*configv1.FeatureGateEnabledDisabled, error) {
+func (o *OperatorOptions) getFeatureGateMappingFromDisk(ctx context.Context, configClient configv1client.Interface) (map[configv1.FeatureSet]*configv1.FeatureGateEnabledDisabled, error) {
+	// TODO get the cluster profile in a better way, this isn't strictly correct
```
Comment (JoelSpeed) [inline pkg/operator/starter.go:219]: How does CVO work it out? Is this the same mechanism?

Comment (deads2k) [reply pkg/operator/starter.go:219]: > How does CVO work it out? Is this the same mechanism?

CVO gets it via an env var that no other pod gets.

**Comment by JoelSpeed** [comment]
/lgtm

Assuming all tests pass

/hold in case timed release required

**Comment by deads2k** [comment]
ExternalOIDC is properly set so the hypershift default is contained as expected.

/hold cancel

**Pull request merged**